packages/app/obojobo-repository/server/models/drafts_metadata.test.js

```diff
@@ -0,0 +1,79 @@
+describe('DraftsMetadata Model', () => {
+	jest.mock('obojobo-express/server/db')
+	jest.mock('obojobo-express/server/logger')
+	let db
+	let logger
+	let DraftsMetadata
+
+	const mockRawDraftsMetadata = {
+		draft_id: 'mockDraftId',
+		key: 'key',
+		value: 'value'
+	}
+
+	const expectMatchesRawMock = draftMetadata => {
+		expect(draftMetadata.draftId).toBe('mockDraftId')
+		expect(draftMetadata.createdAt).toBeUndefined()
+		expect(draftMetadata.updatedAt).toBeUndefined()
+		expect(draftMetadata.key).toBe('key')
+		expect(draftMetadata.value).toBe('value')
+	}
+
+	beforeEach(() => {
+		jest.resetModules()
+		jest.resetAllMocks()
+		db = require('obojobo-express/server/db')
+		logger = require('obojobo-express/server/logger')
+
+		DraftsMetadata = require('./drafts_metadata')
+	})
+	afterEach(() => {})
+
+	test('creating a new DraftsMetadata object maps arguments to properties correctly', () => {
+		const draftMetadata = new DraftsMetadata(mockRawDraftsMetadata)
+		expectMatchesRawMock(draftMetadata)
+	})
+
+	test('saveOrCreate returns a DraftsMetadata object', () => {
+		db.none = jest.fn()
+		db.none.mockResolvedValueOnce(mockRawDraftsMetadata)
+
+		const draftMetadata = new DraftsMetadata(mockRawDraftsMetadata)
+
+		expect.hasAssertions()
+
+		// Trying to match whitespace with the query that's actually running
+		const query = `
```

Comment:
another idea for these (if string contains isn't any good), use a snapshot test.

Reply:
I'm mostly writing out queries like this to pass them into a `toHaveBeenCalledWith` to make sure we're sending the right queries to our `db` calls - maybe it's a feature of snapshots I'm not familiar with, but I'm not sure if they'd be applicable here.

Reply:
Don't forget you can just extract a single argument to a function `db.one.calls[0][0]` (first argument of first call).

Another option is `.toHaveBeenCalledWith(expect. stringContaining('query'), {})`

perhaps `expect(db.one.calls[0][0].replace(/[\W\t\n\r]+/g, ' ')).toBe(expectedQuery.replace(/[\W\t\n\r]+/g, ' '))`

Reply:
That could work. Might want to put it behind reusable function to denastify it for multiple uses, though. Probably a bit late to retrofit the tests now, but perhaps for a refactor later and/or all future tests doing string comparisons like this.